File: Pnut/Pnut/Pnut/Models/Requests/GetTaskRequest.cs
using Pnut.Enums;

namespace Pnut.Models.Requests
{
    public class GetTaskRequest
    {
        public int UserId { get; set; }
        public bool IsGetAllStatus { get; set; }
        public EnumTaskStatus Status { get; set; }
        public bool IsGetAllDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }
}

File: PnutTask/PnutTask/src/Models/Requests/GetTaskRequest.ts
export enum EnumTaskStatus{
    Todo = 0,
    InProgress = 1,
    Pending = 2,
    Done = 3,
    Delete = 4,
    Failed = 5
};

export interface GetTaskRequest{
    UserId: number,
    IsGetAllStatus: boolean,
    Status: EnumTaskStatus,
    IsGetAllDate: boolean,
    StartDate: Date,
    EndDate: Date
};
